File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LowestLayPriceFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.Side;
import org.springframework.stereotype.Component;

@Component
public class LowestLayPriceFunction extends AbstractPriceReduceFunction {

    public LowestLayPriceFunction() {
        super(Side.LAY, Math::min);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LowestLayPriceFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.Side
import org.springframework.stereotype.Component
import kotlin.math.min

@Component
object LowestLayPriceFunction : ProgressFunction by PriceReduceFunction(Side.LAY, { a, b -> min(a, b) })
